configuration/config2/SchemaValue.cpp

```diff
@@ -204,69 +204,68 @@ void SchemaValue::getAllowedValues(std::vector<AllowedValue> &allowedValues, con
     }
 
     //
-    // Is there a specialized rule that limits the values?
-    else if (!m_valueLimitRuleType.empty())
+    // Is there a specialized rule that limits the values? The order is important here. Place more restrictive rules first. Note also that
+    // a rule could be used inside a unique value set which should remain the last entry in this if then else if block
+
+    //
+    // uniqueItemType_1 - value is based on a unique item type described by the data for the rule. This is version 1
+    else if (m_valueLimitRuleType == "uniqueItemType_1")
     {
+        std::vector<std::string> params = splitString(m_valueLimitRuleData, ",");
+
         //
-        // uniqueItemType_1 - value is based on a unique item type described by the data for the rule. This is version 1
-        if (m_valueLimitRuleType == "uniqueItemType_1")
+        // First parameter is the source values for an attribute search. The two parts of the parameter are the path to the
+        // node set where the atttribute, the second part, name is found (not that there may be no entries). Find all the nodes
+        // for the path (parts[0]), then get all of the values for the attribute (parts[1]). This serves as the list of existing
+        // values that are eliminated from the final list of allowable values.
+        ConfigPath sourcePath(params[0]);
```

Comment:
should we avoid lines 222 on  if params[0] is empty, or if params.length < 2?

Reply:
I added error checking for the number of parameters parsed. If the number does not equal the expected results, an exception is thrown.